packages/react-jss/src/utils/memoizeOne.js

```diff
@@ -6,28 +6,18 @@ type ComputeFn = (...args: Args) => Result
 type MemoFn = (...args: Args) => Result
 
 const memoize = (fn: ComputeFn): MemoFn => {
-  let lastArgs
-  let lastResult
+  const cache = new Map<string, Result>()
 
   return (...args: Args): Result => {
-    if (Array.isArray(lastArgs) && args.length === lastArgs.length) {
-      let isSame = true
-
-      for (let i = 0; i < args.length; i++) {
-        if (args[i] !== lastArgs[i]) {
-          isSame = false
-        }
-      }
-
-      if (isSame) {
-        return lastResult
-      }
+    const key: string = JSON.stringify(args)
```

Comment:
this is very questionable, you don't know if the data is serializable

Reply:
Instead of serializing this, can you use some hashing function (such as [object-hash](https://www.npmjs.com/package/object-hash))?

Reply:
Yes, but we used it where we sent an array

Reply:
Also, all tests passed

Reply:
> Instead of serializing this, can you use some hashing function (such as [object-hash](https://www.npmjs.com/package/object-hash))?

Why do we have to add a 10KB package for that?

Reply:
> this is very questionable, you don't know if the data is serializable

Yes, but we know where we used it

Reply:
You guys are already using Terser in your rollup.config, so it's probably not a big deal.

Reply:
> You guys are already using Terser in your rollup.config, so it's probably not a big deal.

I know, but is that less than 14 bytes? also `JSON.stringify` is a native method, it is faster than other solution

Reply:
Aim of `object-hash` is different is not good solution for that

Reply:
@kof What do you think about `object-hash`? I disagree

Reply:
I just checked, we only use memoizeOne to memoize classes object, that means
1. we know it is flat
2. we know it is serializable 

Sorry for confusion, I haven't looked up before where it isused as I wrote that message

Reply:
No problems, so I will resolve this conversation

Reply:
btw. the reason I got confused is that I though we use it in other places too, so if we are to deploy this assumptions, we have to rename this function to something that reflects them

Reply:
Exactly, we need a new PR for that

Reply:
haha no, if you modify what function does, you also have to rename it in the same PR, but there is no problem here.

Reply:
Why are you laghing? :) ok, I will rename this file to `memoize`, are you agree?